packages/salesforcedx-vscode-core/src/commands/forceApexExecute.ts

```diff
@@ -4,99 +4,175 @@
+      channelService.appendLine(e.message);
+    }
+  }
+
+  public executeWrapper(
+    fn: (...args: any[]) => Promise<ExecuteAnonymousResponse>
+  ) {
+    const commandName = this.executionName;
+
+    return async function(...args: any[]): Promise<ExecuteAnonymousResponse> {
+      channelService.showCommandWithTimestamp(`Starting ${commandName}`);
+
+      const result = await vscode.window.withProgress(
+        {
+          title: commandName,
+          location: vscode.ProgressLocation.Notification
+        },
+        async () => {
+          // @ts-ignore
+          return (await fn.call(this, ...args)) as ExecuteAnonymousResponse;
+        }
+      );
 
-export function writeFileAsync(fileName: string, inputText: string) {
-  return new Promise((resolve, reject) => {
-    fs.writeFile(fileName, inputText, err => {
-      if (err) {
-        reject(err);
+      const formattedResult = formatResult(result);
+      channelService.appendLine(formattedResult);
+      channelService.showCommandWithTimestamp(`Finished ${commandName}`);
+
+      if (result.result.compiled && result.result.success) {
+        ApexLibraryExecuteExecutor.errorCollection.clear();
+        await notificationService.showSuccessfulExecution(commandName);
       } else {
-        resolve(true);
+        const editor = vscode.window.activeTextEditor;
+        const document = editor!.document;
+        const filePath = args[0].apexFilePath || document.uri.fsPath;
+
+        handleApexLibraryDiagnostics(
+          result,
+          ApexLibraryExecuteExecutor.errorCollection,
+          filePath
+        );
+        notificationService.showFailedExecution(commandName);
       }
-    });
-  });
+
+      return result;
+    };
+  }
 }
 
-const workspaceChecker = new SfdxWorkspaceChecker();
-const fileNameGatherer = new CreateApexTempFile();
+export function formatResult(
+  execAnonResponse: ExecuteAnonymousResponse
+): string {
+  let outputText: string = '';
+  if (execAnonResponse.result.compiled === true) {
+    outputText += `${nls.localize('apex_execute_compile_success')}\n`;
+    if (execAnonResponse.result.success === true) {
+      outputText += `${nls.localize('apex_execute_runtime_success')}\n`;
+    } else {
+      outputText += `Error: ${execAnonResponse.result.exceptionMessage}\n`;
+      outputText += `Error: ${execAnonResponse.result.exceptionStackTrace}\n`;
+    }
+    outputText += `\n${execAnonResponse.result.logs}`;
+  } else {
+    outputText += `Error: Line: ${execAnonResponse.result.line}, Column: ${
+      execAnonResponse.result.column
+    }\n`;
+    outputText += `Error: ${execAnonResponse.result.compileProblem}\n`;
+  }
+  return outputText;
+}
 
-export async function forceApexExecute(withSelection?: any) {
+export async function forceApexExecute() {
   const commandlet = new SfdxCommandlet(
     workspaceChecker,
     fileNameGatherer,
-    new ForceApexExecuteExecutor()
+    new ApexLibraryExecuteExecutor()
```

Comment:
This might not be the right place for it but we should have a setting that would allow users to use the old CLI implementation, at least while we roll out the library changes.

Reply:
Yeah I'll open up a follow up PR for that